### ISSUE TITLE
suggestion for solving the problem of Player's arms blocking the view

### ISSUE THREAD

**Issue by peow** [tr7zw/FirstPersonModel#441]
There are some moments when being able to see your arms ends up being a nuisance, such as when you are aiming upwards using a crossbow (to hit a flying mob like the Phantom or Ender Dragon)

![2024-04-12_12 36 37](https://github.com/tr7zw/FirstPersonModel/assets/105888751/675e6b69-f945-4106-b257-2985bcd541e0)
![2024-04-12_13 09 53](https://github.com/tr7zw/FirstPersonModel/assets/105888751/94d9b310-6813-471d-8a4b-81d51ebff2e0)

-----------------

What is the **solution** I imagine: when the player's view gets blocked by their arms, the mod would recognize this and adjust the opacity of the player's model and the held item, increasing their transparency, as I did in this image edit:

![1](https://github.com/tr7zw/FirstPersonModel/assets/105888751/7f59d895-76b0-46d1-aa35-6b971f04e2ac)
![2](https://github.com/tr7zw/FirstPersonModel/assets/105888751/1a8c7e5d-ed30-46af-b237-da0a35701234)

----------

This would be a problem sometimes, right? because whenever the arm appeared in the player's view it would be transparent;
to avoid this, I thought this function would only be activated when the Player is **looking up**, and is holding down the RMB (or has the crossbow ready to shoot)

but this would not solve, for example, situations in which the item obstructs the player's view when he is not looking up, such as the Slingshot from the Supplementaries mod
![2024-04-12_15 52 36](https://github.com/tr7zw/FirstPersonModel/assets/105888751/475983b6-8f37-44e9-8dbb-87a995d3f5de)

to solve this, I imagine it would be interesting to make the item more transparent when it approaches the **center** of the screen (while holding RMB)

![3](https://github.com/tr7zw/FirstPersonModel/assets/105888751/14395b42-994b-4d4e-aeee-c16ab5f38697)

-------------------

I don't know if it's possible to implement this, I tried to search if anyone had already posted an Issue about this problem but I couldn't find it, sorry if it's a duplicate post

**Comment by tr7zw** [comment]
You basically can't know when the arms themselves are blocking the view without a ton of math, and for the items is practically not possible. Added to that, transparency wouldn't work well. #437 added a decent middle ground solution(second video).

**Comment by tr7zw** [comment]
> Isn't there a way to know if something is close to the center of the screen in first person?

Not with a ton of math and just guesswork, things I won't do.

> does the feature Dynamic Hands only work when aiming down or also up?

It shows the normal vanilla view, and dynamically switches to the firstperson view when looking down.
